### PR TITLE
update Clang easyblock to add support for building extra tools + leveraging hwloc and Z3 as optional dependencies

### PR DESCRIPTION
**Pull request by geimer**
Add support for building the extra Clang tools, and optionally configuring use of `hwloc` by the OpenMP runtime, and `Z3` bei the Clang static analyzer.

### CODE REVIEW

**Review by ocaisa — APPROVED** [review]
LGTM, might be a while before generoso can build and test this on the open easyconfig PR

File: easybuild/easyblocks/c/clang.py [hunk 1]
```diff
@@ -162,7 +165,10 @@ def find_source_dir(globpatterns, targetdir):
             find_source_dir('libcxx-*', os.path.join(self.llvm_src_dir, 'projects', 'libcxx'))
             find_source_dir('libcxxabi-*', os.path.join(self.llvm_src_dir, 'projects', 'libcxxabi'))
 
-        find_source_dir(['clang-*', 'cfe-*'], os.path.join(self.llvm_src_dir, 'tools', 'clang'))
+        find_source_dir(['clang-[1-9]*', 'cfe-*'], os.path.join(self.llvm_src_dir, 'tools', 'clang'))
+
+        if self.cfg["build_extra_clang_tools"]:
+            find_source_dir('clang-tools-extra-*', os.path.join(self.llvm_src_dir, 'tools/clang/tools', 'extra'))
```
Comment (ocaisa) [inline easybuild/easyblocks/c/clang.py:171]: Please use
```
find_source_dir('clang-tools-extra-*', os.path.join(self.llvm_src_dir, 'tools, 'clang', 'tools', 'extra'))
```

**Review by ocaisa — CHANGES_REQUESTED** [review]
Minor comment

File: easybuild/easyblocks/c/clang.py (hunk 1 above)
Comment (boegel) [reply easybuild/easyblocks/c/clang.py]: fixed in 95a5e0887

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Clang-6.0.1-GCC-6.4.0-2.28.eb_ 
 * **SUCCESS** _Clang-9.0.1-GCCcore-8.3.0.eb_ 
 * **SUCCESS** _Clang-11.0.0-GCCcore-9.3.0.eb_ 

Build succeeded for 3 out of 3 (3 easyconfigs in total)
node2606.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/d6647eac4099662c5fe21d21288b2ded for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Clang-6.0.1-GCC-6.4.0-2.28.eb_ 
 * **SUCCESS** _Clang-8.0.0-GCCcore-8.2.0.eb_ 
 * **SUCCESS** _Clang-10.0.0-GCCcore-9.3.0.eb_ 

Build succeeded for 3 out of 3 (3 easyconfigs in total)
node3136.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/daf600641cb99b7c905a7d355ed0615a for a full test report.

**Pull request merged**